dist/common/scripts/scylla_util.py

```diff
@@ -376,7 +376,7 @@ def __init__(self):
     def is_azure_instance():
         """Check if it's Azure instance via DNS lookup to metadata server."""
         try:
-            addrlist = socket.getaddrinfo('metadata.azure.internal', 80)
+            addrlist = socket.getaddrinfo('169.254.169.254', 80)
```

Comment:
this will do a reverse lookup
I don't get why this is needed, the VMs I saw all contain in /etc/hosts proper 
metadata.azure.internal 169.254.169.254
record, which is what we want to test, @avikivity wanted to remove the final check here if it resolved to 169.254.169.254 (to support mocking of Azure)
so now we check in reverse way?
I'd like to get an explanation/reason/argument why this change is needed (what problem are we trying to solve?)

Reply:
@tarzanek So `metadata.azure.internal 169.254.169.254` is not in /etc/hosts so i guess this is the reason it's not working . i will add it as part of the machine-image-installation